### PR TITLE
Change to class variable

### PR DESCRIPTION
**Pull request by t-kameyama**
I think the "p" (key regex pattern) variable should be a class variable.

### CODE REVIEW

File: src/main/java/spark/QueryParamsMap.java [hunk 1]
```diff
@@ -33,6 +33,8 @@
 
     private static final QueryParamsMap NULL = new NullQueryParamsMap();
 
+    private static final Pattern KEY_PATTERN = Pattern.compile("\\A[\\[\\]]*([^\\[\\]]+)\\]*");
```
Comment (dinesh707) [inline src/main/java/spark/QueryParamsMap.java]: can you add a doc comment about this pattern. Then its easy to read in future

File: src/main/java/spark/QueryParamsMap.java [hunk 2]
```diff
@@ -33,6 +33,9 @@
 
     private static final QueryParamsMap NULL = new NullQueryParamsMap();
 
+    /**
+     * Pattern for parsing the key of querystring
+     */
     private static final Pattern KEY_PATTERN = Pattern.compile("\\A[\\[\\]]*([^\\[\\]]+)\\]*");
```
Comment (t-kameyama) [inline src/main/java/spark/QueryParamsMap.java]: @dinesh707 
I added javadoc comment

**Comment by dinesh707** [comment]
+1

File: src/main/java/spark/QueryParamsMap.java [hunk 3]
```diff
@@ -34,6 +34,11 @@
     private static final QueryParamsMap NULL = new NullQueryParamsMap();
 
     /**
+     * Pattern for parsing the key of querystring
+     */
+    private static final Pattern KEY_PATTERN = Pattern.compile("\\A[\\[\\]]*([^\\[\\]]+)\\]*");
```
Comment (perwendel) [inline src/main/java/spark/QueryParamsMap.java]: I'm not the author of this code but as my understanding goes the matcher that's fetched from the pattern instance has an internal state. This change means that all requests would share that state. Doesn't feel right.

**Comment by t-kameyama** [comment]
@perwendel 
The Patten class is thread safe.
The Patten#matcher method returns new Matcher instance.
So, All requests does not share the state.

The purpose of this change is to reduce the compile cost of the Pattern.

**Comment by perwendel** [comment]
@t-kameyama 
Yep, you're correct. It's the matcher that shouldn't be shared between threads but we're not doing that!

**Pull request merged**